sky/spot/controller.py

```diff
@@ -61,12 +61,22 @@ def __init__(self, job_id: int, task_yaml: str,
     def _run(self):
         """Busy loop monitoring spot cluster status and handling recovery.
 
-        Raises:
-            exceptions.ResourcesUnavailableError: if the spot cluster fails
-                to be launched or the job fails to be submitted to the cluster.
-                This will happen iff none of the failover are due to resources
-                unavailability, or retry_until_up is False and we've reached the
-                maximum number of retries.
+        Raises (non-exhaustive):
+            exceptions.ProvisionPrechecksError: This will be raised when the
+                underlying `sky.launch` fails due to any error happens during
+                the prechecks for all the clouds, i.e. none of the failover, if
+                any, is due to the resources unavailability. It includes the
+                following cases:
+                1. The optimizer cannot find a feasible solution.
+                2. Invalid cluster name, failure in getting cloud user identity,
+                or unsupported feature happens.
+            exceptions.SpotJobReachedMaxRetryError: This will be raised when the
+                maximum number of retries is reached for `sky.launch`. The
+                failure of `sky.launch` can be due to:
+                1. Any of the underyling failover is due to the resources
+                unavailability.
+                2. The cluster is preempted before the job is submitted.
+                3. Any unexpected error happens during the `sky.launch`.
```

Comment:
```suggestion
            exceptions.ProvisionPrechecksError: This will be raised when the
                underlying `sky.launch` fails due to precheck errors only.
                I.e., none of the failover exceptions, if
                any, is due to resources unavailability. This exception includes the
                following cases:
                1. The optimizer cannot find a feasible solution.
                2. Precheck errors: invalid cluster name, failure in getting cloud user identity, or unsupported feature.
            exceptions.SpotJobReachedMaxRetryError: This will be raised when all prechecks passed but the
                maximum number of retries is reached for `sky.launch`. The
                failure of `sky.launch` can be due to:
                1. Any of the underlying failover exceptions is due to resources
                unavailability.
                2. The cluster is preempted before the job is submitted.
                3. Any unexpected error happens during the `sky.launch`.
```
Did some rewording; may need reformat.